provider.go

```diff
@@ -140,33 +142,38 @@ func New(options ...ProviderHandler) (*WasmcloudProvider, error) {
 		putTargetLinkFunc: func(InterfaceLinkDefinition) error { return nil },
 		delSourceLinkFunc: func(InterfaceLinkDefinition) error { return nil },
 		delTargetLinkFunc: func(InterfaceLinkDefinition) error { return nil },
+		sourceLinkArr:     sourceLinks,
+		targetLinkArr:     targetLinks,
```

Comment:
Instead of having a separate map, I think we can keep storing these just as `source/targetLinks` and initialize them later